packages/@tinacms/api-git/src/file-writer.ts

```diff
@@ -25,13 +25,26 @@ let count = 0
 
 const MAX_BUILD_TIME = 1000
 
-export function writeFile(filepath: string, content: string) {
-  count++
+export function checkFilePathIsInGitRepo(filepath: string, repoAbsolutePath: string) {
+  const fullpath = path.resolve(filepath)
+  const repopath = path.resolve(repoAbsolutePath)
+  if (fullpath.includes(repopath)) {
+    return true
+  } else {
+    return false
+  }
+}
 
+export function writeFile(filepath: string, content: string, repoAbsolutePath: string) {
+  count++
+  if (DEBUG) console.info(`checkFilePathIsInGitRepo? ${checkFilePathIsInGitRepo(filepath, repoAbsolutePath)}`)
   if (DEBUG) console.info(`request ${count} received`)
-
-  cacheCommand(filepath, content)
-  tryToWrite()
+  if (checkFilePathIsInGitRepo(filepath, repoAbsolutePath)) {
```

Comment:
I would prefer to move `checkFilePathIsInGitRepo` out of `file-writer.ts` and into `router.ts`

Currently `writeFile` is a low level API in the domain of the filesystem. It just _eventually_ writes the given string to the given path. 

This PR introduces the concept of a git repository to `writeFile`

Instead, undo the changes here and move the check up so that `router.ts` looks something like:

```typescript
    try {
      const fileIsInRepo = checkFilePathIsInGitRepo(
        fileAbsolutePath,
        REPO_ABSOLUTE_PATH
      )
       
      if (fileIsInRepo) {
        writeFile(fileAbsolutePath, req.body.content)
      }
      res.json({ content: req.body.content })
    } catch (e) {

Reply:
ok, I wil move into router.